Jolt/Core/Core.h

```diff
@@ -42,9 +42,13 @@
 #endif
 
 // Detect CPU architecture
-#if defined(__x86_64__) || defined(_M_X64)
+#if defined(__x86_64__) || defined(_M_X64) || defined(__i386__) || defined(_M_IX86)
 	// X86 CPU architecture
-	#define JPH_CPU_X64
```

Comment:
I think we should probably just rename JPH_CPU_X64 to JPH_CPU_X86 and have a separate define for JPH_CPU_ADDRESS_BITS (= 32 or 64, not entirely happy with the name but I can't come up with a better one)

Reply:
What are your thoughts on only having JPH_CPU_X86 and using `sizeof(void*) == X` + ternaries  given it's consteval?

I myself am not a big macro fan, but if you'd prefer that we can do that.

Reply:
Since all platform specifics are already done using defines, I'd keep it a define for consistency.